### PR TITLE
feat: Game end event

### PR DESCRIPTION
**Pull request by 318097**
Based on Issue #126, the game ends when,
* Both the players are dead
* All the food is eaten

There is a problem in sending the event, as 'io' is undefined. Whoever is responsible, please look into this matter.

### CODE REVIEW

File: server/api/game/controller.js [hunk 1]
```diff
@@ -135,12 +137,36 @@ class Game {
...
+
   startGame = () => {
     this.setInitialGameState();
     // eslint-disable-next-line no-console
     console.log('Game started');
     this.gameState.interval = setInterval(() => {
       this.calculateNextGameState();
+      const gameEndStatus = this.hasGameEnded();
+      if (gameEndStatus) {
+        // io.in(this.roomId).emit(GAME_OVER);
```
Comment (Shalinit3) [inline server/api/game/controller.js]: Why is this line commented?

File: server/api/game/controller.js [hunk 2]
```diff
@@ -135,12 +137,36 @@ class Game {
...
   };
 
+  hasGameEnded = () => {
+    const { players } = this.gameState;
+    const playerStatus = Object.keys(players).map(player => player.alive);
+    const { isPlayerOneAlive, isPlayerTwoAlive } = playerStatus;
+    return this.hasFoodFinished() || (!isPlayerOneAlive && !isPlayerTwoAlive);
+  };
+
+  hasFoodFinished = () => {
```
Comment (Shalinit3) [inline server/api/game/controller.js]: This could be moved to core.js

File: server/api/game/controller.js (hunk 1 above)
Comment (318097) [reply server/api/game/controller.js]: Because I didn't know that `io` is global.

**Review by shivangg — APPROVED** [review]
What is the difference between GAME_END and GAME_OVER?

**Pull request merged**